### ISSUE TITLE
Make it possible to skip either frontend or backend CI jobs 

### ISSUE THREAD

**Issue by Abbe98** [OpenRefine/OpenRefine#5755]
A lot of pull requests either affect the backend or frontend but currently one can just skip all of CI or none, which is especially annoying when multiple jobs gets queued.

### Proposed solution
We could add two commit message keywords and conditions to frontend respective backend related tests/quality checks:

```
if: ${{ !contains(github.event.head_commit.message, '#frontend-only') }}
```

```
if: ${{ !contains(github.event.head_commit.message, '#backend-only') }}
```

So that contributors can skip jobs that aren't relevant to their contribution. All jobs would still run on merges.

**Comment by Abbe98** [comment]
Random reference on the Internet: https://www.eliostruyf.com/devhack-skip-github-actions-specific-commits-messages/

**Comment by thadguidry** [comment]
Maybe instead of putting a small burden on contributors to add something to their commit, we could take advantage of known certain paths and if files are changed within those paths, we can ignore to do some workflows.  We did that with the docs.  

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-paths
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

In general, we are doing too many different things in the single 1 `pull_request.yml` file and instead we should have 2 or 3 different other workflows.  We could have one for just building and testing backend with Java.  Another for just frontend build setup and testing the frontend.

Besides having more independent workflows, within them we could also take advantage of more of the provided workflow syntax that is now available after the last 4 years, such as `needs` https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow  as well as using conditionals like `if` https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#further-controlling-how-your-workflow-will-run Ex.
```
jobs:
  run_if:
    if:  startsWith(github.head_ref, 'releases/')
```
and reusing smaller workflows and calling them within other workflows https://docs.github.com/en/actions/using-workflows/reusing-workflows

In summary, we should start by writing down the parts of CI we have, and then do the work to break those up and partition those into more workflows, then jobs (which can run concurrently), then steps.

**Comment by tfmorris** [comment]
One reason for caution is that the "front end" tests are really the only form of system/integration tests that OpenRefine has, so you really want them to run on all backend changes too.

**Comment by Abbe98** [comment]
> One reason for caution is that the "front end" tests are really the only form of system/integration tests that OpenRefine has, so you really want them to run on all backend changes too.

Oh yes, I would expect these to run whenever the backend or frontend changes.
